ktlint raise error when there is empty enum class

<!-- The bug you're experiencing might have already be reported!
Please search in the [issues](https://github.com/pinterest/ktlint/issues) before creating one. -->

## Expected Behavior
<!---Tell us what should happen. -->
<!-- If relevant, provide a link to the relevant section in the [Kotlin Coding Conventions](https://kotlinlang.org/docs/coding-conventions.html) or [Android Kotlin Style Guide](https://developer.android.com/kotlin/style-guide).  -->

<!-- Note: Ktlint should not produce any code which conflict with IntelliJ default code formatting. So please verify that your expectation is accepted by IntelliJ default code formatting. -->
When kotlin project has empty enum class, ktlint raise error.

```
$ git clone https://github.com/ganadist/VersionCodeDemo -b ktlint_0471_error
$ cd VersionCodeDemo
$ ./gradlew :ktlint
```

## Observed Behavior
<!---Tell us what happens instead of the expected behavior -->
<!--- Provide the exact command which was executed but please -->
<!--- ensure that the flag "--log-level=debug" is added to the -->
<!--- command as well. Provide the output of this command. -->
```
> Task :ktlint FAILED
00:41:59.889 [main] DEBUG com.pinterest.ktlint.internal.KtlintCommandLine - Discovered reporter with "baseline" id.
00:41:59.890 [main] DEBUG com.pinterest.ktlint.internal.KtlintCommandLine - Discovered reporter with "checkstyle" id.
00:41:59.890 [main] DEBUG com.pinterest.ktlint.internal.KtlintCommandLine - Discovered reporter with "json" id.
00:41:59.890 [main] DEBUG com.pinterest.ktlint.internal.KtlintCommandLine - Discovered reporter with "format" id.
00:41:59.890 [main] DEBUG com.pinterest.ktlint.internal.KtlintCommandLine - Discovered reporter with "html" id.
00:41:59.890 [main] DEBUG com.pinterest.ktlint.internal.KtlintCommandLine - Discovered reporter with "plain" id.
00:41:59.890 [main] DEBUG com.pinterest.ktlint.internal.KtlintCommandLine - Discovered reporter with "sarif" id.
00:41:59.890 [main] DEBUG com.pinterest.ktlint.internal.KtlintCommandLine - Initializing "plain" reporter with {verbose=false, color=false, color_name=DARK_GRAY, format=false}
00:41:59.904 [main] DEBUG com.pinterest.ktlint.internal.FileUtils - Start walkFileTree for rootDir: 'VersionCodeDemo'
   include:
[      - sun.nio.fs.UnixFileSystem$3@2e4b8173]
   exclude:
[      - sun.nio.fs.UnixFileSystem$3@70e8f8e,       - sun.nio.fs.UnixFileSystem$3@17046283,       - sun.nio.fs.UnixFileSystem$3@5bd03f44,       - sun.nio.fs.UnixFileSystem$3@29626d54,       - sun.nio.fs.UnixFileSystem$3@5a63f509,       - sun.nio.fs.UnixFileSystem$3@6e4784bc,       - sun.nio.fs.UnixFileSystem$3@34b7ac2f,       - sun.nio.fs.UnixFileSystem$3@e056f20]
00:41:59.913 [main] DEBUG com.pinterest.ktlint.internal.FileUtils - Discovered 1 files to be processed in 9 ms
00:42:00.357 [pool-1-thread-1] DEBUG com.pinterest.ktlint.core.internal.RuleRunnerSorter - Rule with id 'standard:indent' should run after the rule with id 'experimental:function-signature'. However, the latter rule is not loaded and is allowed to be ignored. For best results, it is advised load the rule.
00:42:00.358 [pool-1-thread-1] DEBUG com.pinterest.ktlint.core.internal.RuleRunnerSorter - Rules will be executed in order below (unless disabled):
           - standard:annotation, 
           - standard:annotation-spacing, 
           - standard:argument-list-wrapping, 
           - standard:chain-wrapping, 
           - standard:colon-spacing, 
           - standard:comma-spacing, 
           - standard:comment-spacing, 
           - standard:curly-spacing, 
           - standard:dot-spacing, 
           - standard:double-colon-spacing, 
           - standard:enum-entry-name-case, 
           - standard:filename, 
           - standard:final-newline, 
           - standard:import-ordering, 
           - standard:keyword-spacing, 
           - standard:modifier-order, 
           - standard:multiline-if-else, 
           - standard:no-blank-line-before-rbrace, 
           - standard:no-blank-lines-in-chained-method-calls, 
           - standard:no-consecutive-blank-lines, 
           - standard:no-empty-class-body, 
           - standard:no-empty-first-line-in-method-block, 
           - standard:no-line-break-after-else, 
           - standard:no-line-break-before-assignment, 
           - standard:no-multi-spaces, 
           - standard:no-semi, 
           - standard:no-trailing-spaces, 
           - standard:no-unit-return, 
           - standard:no-unused-imports, 
           - standard:no-wildcard-imports, 
           - standard:op-spacing, 
           - standard:package-name, 
           - standard:parameter-list-wrapping, 
           - standard:paren-spacing, 
           - standard:range-spacing, 
           - standard:spacing-around-angle-brackets, 
           - standard:spacing-between-declarations-with-annotations, 
           - standard:spacing-between-declarations-with-comments, 
           - standard:string-template, 
           - standard:unary-op-spacing, 
           - standard:wrapping, 
           - standard:indent, 
           - standard:trailing-comma-on-call-site, 
           - standard:max-line-length, 
           - standard:trailing-comma-on-declaration-site
00:42:00.407 [pool-1-thread-1] DEBUG com.pinterest.ktlint.internal.KtlintCommandLine - Internal Error (trailing-comma-on-declaration-site) in file '/home/ganadist/src/build/AndroidStudioProjects/vcd/lib/hostconfig/src/main/java/com/example/myapplication/hostconfig/TestView.kt' at position '1:1
com.pinterest.ktlint.core.RuleExecutionException: com.pinterest.ktlint.core.RuleExecutionException: com.pinterest.ktlint.core.RuleExecutionException: com.pinterest.ktlint.core.RuleExecutionException: java.util.NoSuchElementException: Sequence contains no element matching the predicate.
	at com.pinterest.ktlint.core.KtLint.executeRuleOnNodeRecursively(KtLint.kt:233)
	at com.pinterest.ktlint.core.KtLint.executeRule(KtLint.kt:195)
	at com.pinterest.ktlint.core.KtLint.access$executeRule(KtLint.kt:30)
	at com.pinterest.ktlint.core.KtLint$lint$1.invoke(KtLint.kt:176)
	at com.pinterest.ktlint.core.KtLint$lint$1.invoke(KtLint.kt:175)
	at com.pinterest.ktlint.core.internal.VisitorProvider$visitor$6.invoke(VisitorProvider.kt:80)
	at com.pinterest.ktlint.core.internal.VisitorProvider$visitor$6.invoke(VisitorProvider.kt:78)
	at com.pinterest.ktlint.core.KtLint.lint(KtLint.kt:175)
	at com.pinterest.ktlint.internal.FileUtilsKt.lintFile(FileUtils.kt:363)
	at com.pinterest.ktlint.internal.KtlintCommandLine.process(KtlintCommandLine.kt:410)
	at com.pinterest.ktlint.internal.KtlintCommandLine.access$process(KtlintCommandLine.kt:52)
	at com.pinterest.ktlint.internal.KtlintCommandLine$lintFiles$3.invoke$lambda-0(KtlintCommandLine.kt:313)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:829)
Caused by: com.pinterest.ktlint.core.RuleExecutionException: com.pinterest.ktlint.core.RuleExecutionException: com.pinterest.ktlint.core.RuleExecutionException: java.util.NoSuchElementException: Sequence contains no element matching the predicate.
	at com.pinterest.ktlint.core.KtLint.executeRuleOnNodeRecursively(KtLint.kt:233)
	at com.pinterest.ktlint.core.KtLint.executeRuleOnNodeRecursively(KtLint.kt:219)
	... 15 common frames omitted
Caused by: com.pinterest.ktlint.core.RuleExecutionException: com.pinterest.ktlint.core.RuleExecutionException: java.util.NoSuchElementException: Sequence contains no element matching the predicate.
	at com.pinterest.ktlint.core.KtLint.executeRuleOnNodeRecursively(KtLint.kt:233)
	at com.pinterest.ktlint.core.KtLint.executeRuleOnNodeRecursively(KtLint.kt:219)
	... 16 common frames omitted
Caused by: com.pinterest.ktlint.core.RuleExecutionException: java.util.NoSuchElementException: Sequence contains no element matching the predicate.
	at com.pinterest.ktlint.core.KtLint.executeRuleOnNodeRecursively(KtLint.kt:233)
	at com.pinterest.ktlint.core.KtLint.executeRuleOnNodeRecursively(KtLint.kt:219)
	... 17 common frames omitted
Caused by: java.util.NoSuchElementException: Sequence contains no element matching the predicate.
	at com.pinterest.ktlint.ruleset.standard.TrailingCommaOnDeclarationSiteRule.findNodeAfterTrailingCommaPosition(TrailingCommaOnDeclarationSiteRule.kt:505)
	at com.pinterest.ktlint.ruleset.standard.TrailingCommaOnDeclarationSiteRule.visitClass(TrailingCommaOnDeclarationSiteRule.kt:205)
	at com.pinterest.ktlint.ruleset.standard.TrailingCommaOnDeclarationSiteRule.beforeVisitChildNodes(TrailingCommaOnDeclarationSiteRule.kt:88)
	at com.pinterest.ktlint.core.KtLint.executeRuleOnNodeRecursively(KtLint.kt:209)
	... 18 common frames omitted
VersionCodeDemo/lib/hostconfig/src/main/java/com/example/myapplication/hostconfig/TestView.kt:1:1: Internal Error (trailing-comma-on-declaration-site) in file 'VersionCodeDemo/lib/hostconfig/src/main/java/com/example/myapplication/hostconfig/TestView.kt' at position '1:1. Please create a ticket at https://github.com/pinterest/ktlint/issues (if possible, please re-run with the --debug flag to get the stacktrace and provide the source code that triggered an error) (cannot be auto-corrected) ()
00:42:00.409 [main] DEBUG com.pinterest.ktlint.internal.KtlintCommandLine - 555ms / 1 file(s) / 1 error(s)

```

## Steps to Reproduce
<!--- Provide a code example, or an unambiguous set of steps to -->
<!--- reproduce this bug. Minimize the example to the bare -->
<!--- minimum required to reproduce the bug. Small, well written -->
<!--- example are proven to be resolved quicker. -->

* perform ktlint 0.47.1 or master branch with https://github.com/ganadist/VersionCodeDemo/blob/ktlint_0471_error/lib/hostconfig/src/main/java/com/example/myapplication/hostconfig/TestView.kt#L23-L25 code


## Your Environment
<!--- Include as many relevant details about the environment you experienced the bug in -->
* Version of ktlint used: 0.47.1 / 0.48.0-SNAPSHOT
* Relevant parts of the `.editorconfig` settings : https://github.com/ganadist/VersionCodeDemo/blob/ktlint_0471_error/.editorconfig  
* Name and version (or code for custom task) of integration used (Gradle plugin, Maven plugin, command line, custom Gradle task): https://github.com/ganadist/VersionCodeDemo/blob/ktlint_0471_error/gradle/ktlint.gradle
* Version of Gradle used (if applicable): 7.5.1
* Operating System and version: ArchLinux / OSX 13.0

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -213,6 +213,7 @@ Like before, the API Consumer can still offer a mix of rules originating from `k
 * Disable the `standard:filename` rule whenever Ktlint CLI is run with option `--stdin` ([#1742](https://github.com/pinterest/ktlint/issues/1742))
 * The parameters of a function literal containing a multiline parameter list are aligned with first parameter whenever the first parameter is on the same line as the start of that function literal (not allowed in `ktlint_official` code style) `indent` ([#1756](https://github.com/pinterest/ktlint/issues/1756)).
 * Fix continuation indent for a dot qualified array access expression in `ktlint_official` code style only `indent`  ([#1740](https://github.com/pinterest/ktlint/issues/1540)).
+* Do not throw exception when enum class does not contain entries `trailing-comma-on-declaration-site` ([#1711](https://github.com/pinterest/ktlint/issues/1711))
 
 ### Changed
 * Wrap the parameters of a function literal containing a multiline parameter list (only in `ktlint_official` code style) `parameter-list-wrapping` ([#1681](https://github.com/pinterest/ktlint/issues/1681)).

--- a/ktlint-ruleset-standard/src/main/kotlin/com/pinterest/ktlint/ruleset/standard/rules/TrailingCommaOnDeclarationSiteRule.kt
+++ b/ktlint-ruleset-standard/src/main/kotlin/com/pinterest/ktlint/ruleset/standard/rules/TrailingCommaOnDeclarationSiteRule.kt
@@ -198,34 +198,22 @@ public class TrailingCommaOnDeclarationSiteRule :
             .takeIf { psi.isEnum() }
             ?.findChildByType(ElementType.CLASS_BODY)
             ?.takeUnless {
-                // Do nothing when last two entries are on same line as no trailing comma should be inserted
-                it.lastTwoEnumEntriesAreOnSameLine()
+                it.noEnumEntries() || it.lastTwoEnumEntriesAreOnSameLine()
             }?.let { classBody ->
-                val nodeAfterTrailingCommaPosition = classBody.findNodeAfterTrailingCommaPosition()
-                node.reportAndCorrectTrailingCommaNodeBefore(
-                    inspectNode = nodeAfterTrailingCommaPosition,
-                    isTrailingCommaAllowed = node.isTrailingCommaAllowed(),
-                    autoCorrect = autoCorrect,
-                    emit = emit,
-                )
+                classBody
+                    .findNodeAfterLastEnumEntry()
+                    ?.let { nodeAfterLastEnumEntry ->
+                        node.reportAndCorrectTrailingCommaNodeBefore(
+                            inspectNode = nodeAfterLastEnumEntry,
+                            isTrailingCommaAllowed = node.isTrailingCommaAllowed(),
+                            autoCorrect = autoCorrect,
+                            emit = emit,
+                        )
+                    }
             }
     }
 
-    /**
-     * Determines the [ASTNode] before which the trailing comma is allowed.
-     *
-     * If the list of enumeration entries is terminated by a semicolon, that semicolon will be returned. Otherwise, the
-     * last element of the class.
-     */
-    private fun ASTNode.findNodeAfterTrailingCommaPosition(): ASTNode {
-        val lastEnumEntry = children().last { it.psi is KtEnumEntry }
-
-        val semicolonAfterLastEnumEntry = lastEnumEntry
-            .children()
-            .singleOrNull { it.elementType == ElementType.SEMICOLON }
-
-        return semicolonAfterLastEnumEntry ?: lastChildNode
-    }
+    private fun ASTNode.noEnumEntries() = children().none { it.psi is KtEnumEntry }
 
     private fun ASTNode.lastTwoEnumEntriesAreOnSameLine(): Boolean {
         val lastTwoEnumEntries =
@@ -236,6 +224,19 @@ public class TrailingCommaOnDeclarationSiteRule :
 
         return lastTwoEnumEntries.count() == 2 && noNewLineInClosedRange(lastTwoEnumEntries[0], lastTwoEnumEntries[1])
     }
+
+    /**
+     * Determines the [ASTNode] before which the trailing comma is allowed.
+     *
+     * If the list of enumeration entries is terminated by a semicolon, that semicolon will be returned. Otherwise, the
+     * last element of the class.
+     */
+    private fun ASTNode.findNodeAfterLastEnumEntry() =
+        children()
+            .lastOrNull { it.psi is KtEnumEntry }
+            ?.children()
+            ?.singleOrNull { it.elementType == ElementType.SEMICOLON }
+            ?: lastChildNode
 
     private fun ASTNode.reportAndCorrectTrailingCommaNodeBefore(
         inspectNode: ASTNode,

--- a/ktlint-ruleset-standard/src/test/kotlin/com/pinterest/ktlint/ruleset/standard/rules/TrailingCommaOnDeclarationSiteRuleTest.kt
+++ b/ktlint-ruleset-standard/src/test/kotlin/com/pinterest/ktlint/ruleset/standard/rules/TrailingCommaOnDeclarationSiteRuleTest.kt
@@ -1033,4 +1033,17 @@ class TrailingCommaOnDeclarationSiteRuleTest {
             .hasNoLintViolationsForRuleId("no-semi")
             .isFormattedAs(formattedCode)
     }
+
+    @Test
+    fun `Issue 1711 - Given an empty enum class then do not throw an exception`() {
+        val code =
+            """
+            enum class Dummy(val type: String) {
+                // empty
+            }
+            """.trimIndent()
+        trailingCommaOnDeclarationSiteRuleAssertThat(code)
+            .withEditorConfigOverride(TRAILING_COMMA_ON_DECLARATION_SITE_PROPERTY to true)
+            .hasNoLintViolations()
+    }
 }

